### PR TITLE
Api to get iterable index from the DagStore

### PR DESCRIPTION
**Pull request by aarshkshah1992**
We should be able to get an iterable index out of the dagstore so the reference provider can serve the index to the network indexer.

### CODE REVIEW

File: dagstore.go [hunk 1]
```diff
@@ -204,7 +208,7 @@ func NewDAGStore(cfg Config) (*DAGStore, error) {
 	}
 
 	if cfg.InvertedIndex == nil {
-		log.Info("using in-memory inverted index")
+		log.Warn("using in-memory inverted index")
```
Comment (raulk) [inline dagstore.go:211]: Why warn?

File: dagstore.go [hunk 2]
```diff
@@ -333,10 +337,28 @@ func (d *DAGStore) Start(ctx context.Context) error {
 	return nil
 }
 
+func (d *DAGStore) GetIterableIndex(key shard.Key) (carindex.IterableIndex, error) {
+	fi, err := d.indices.GetFullIndex(key)
+	if err != nil {
+		return nil, err
```
Comment (raulk) [inline dagstore.go:343]: Wrap in "failed to get index"

File: dagstore.go [hunk 3]
```diff
@@ -333,10 +337,28 @@ func (d *DAGStore) Start(ctx context.Context) error {
...
+}
+
 func (d *DAGStore) GetShardKeysForCid(c cid.Cid) ([]shard.Key, error) {
 	return d.invertedIndex.GetShardsForMultihash(c.Hash())
 }
 
+func (d *DAGStore) GetShardKeysForMultihash(h mh.Multihash) ([]shard.Key, error) {
+	return d.invertedIndex.GetShardsForMultihash(h)
+}
+
```
Comment (raulk) [inline dagstore.go:361]: Method names are unspecific, as they do not imply what the relationship between the shard and CID/MH is. Please rename to "ShardsContainingCid" and "ShardsContainingMultihash"

(Not sure I like having two variants, either)

**Review by dirkmc — APPROVED** [review]
Modulo raul's suggestions LGTM 👍

File: dagstore.go (hunk 2 above)
Comment (aarshkshah1992) [reply dagstore.go]: Done.

File: dagstore.go (hunk 3 above)
Comment (aarshkshah1992) [reply dagstore.go]: Changed to having only the `ShardsContainingMultihash ` variant as we can always go from a cid -> multihash.

File: dagstore.go (hunk 1 above)
Comment (aarshkshah1992) [reply dagstore.go]: fixed.

**Pull request merged**